### DISCUSSION TITLE
November release - Blocks, Oracle integration, new dynamic filter and more

### DISCUSSION THREAD

**Discussion by joebudi** [Budibase/budibase#3697] [Announcements]
Hello Budis!

In November, Budibase launched out of beta! The response from our community was amazing, and over 5000 new accounts signed up to Budibase. If this wasn’t enough, we’ve found time to release a number of new features:

**Budibase Blocks**
Blocks are predefined sets of components that streamline the design process within Budibase. Before blocks, creating a list of searchable cards took 10 minutes; with blocks, this process takes under 1 minute. 

**Oracle DB Integration**
A popular request amongst our community, we’re happy to announce you can now use Oracle DB to build business apps with Budibase. Going one step further, we’ve made the Oracle datasource a ‘Datasource Plus’ which allows you to generate tables directly from your Oracle database.

**MSSQL Datasource Plus**
Building on our MSSQL integration, you can now generate tables directly from your MSSQL database.

**New Screen Creation UI**
Autogenerate CRUD screens from any datasource with just a few clicks. The new UI and UX stems from numerous conversations with the community and makes the design experience consistent between the internal database and external datasources.

**Filter Component**
You can now include a dynamic filter component inside a data provider, and have complete control over all of the filters - automatically. Users can now create incredible search and filtering experiences with just a few clicks.
![CleanShot_2021-12-02_at_14 50 13_pyvxfw (1)](https://user-images.githubusercontent.com/49767913/144852752-8817cecc-b160-4985-8911-fcadd94afbec.png)

**Add text directly to the design screen**
Type directly onto your design screen instead of using the settings panel. For simple text input, we feel this is a better experience.
![CleanShot_2021-12-02_at_12 46 34_qsf5li (1)](https://user-images.githubusercontent.com/49767913/144852680-34edaf3f-0147-4888-9afa-459365f8c060.gif)

**Additional, important improvements**
- Improved SQL relationships UI - faster and more flexible
- JavaScript formula fields
- Access levels on custom queries
- Import an existing table into a CSV
- Use _id fields in relationships, and have access to attachment information such as the filename of the attachment.
- Use schemas other than “public” in postgres data source plus

**Coming up in the December Release**
- New and improved Rest API experience
- JSON data field
- New Budibase home screen UI
- New templates

Watch the release video [here](https://youtu.be/nq4K69QDfC0)

**Comment by marco-santulli** [comment]
cool! does it update automatically or do we need to do something?

**Reply by joebudi** [reply]
Hey
If you are talking about the dynamic filter - yes it will update automatically.

**Reply by marco-santulli** [reply]
sorry, I mean I am on 0.9.187, normally the release updates automatically from within the web GUI, but it says 0.9.187 is the latest release..
are we supposed to uninstall / install or something like that? or is it not available yet?

**Reply by KristofferBroberg** [reply]
Try `budi hosting --update` in the terminal.

**Reply by marco-santulli** [reply]
I tried, and tried updating from within the GUI, I always see:
<img width="397" alt="image" src="https://user-images.githubusercontent.com/82715690/144936441-4c33a3f9-8954-41b0-8fc3-acc373413b4b.png">

**Reply by SantoshSrinivas79** [reply]
@marco-santulli I faced the same issue ... resolution [here](https://github.com/Budibase/budibase/discussions/3574#discussioncomment-1738192)

**Reply by starmomou** [reply]
@marco-santulli 
`docker-compose --env-file hosting.properties pull && docker-compose --env-file hosting.properties up -d`

**Reply by marco-santulli** [reply]
Hi, that didn't work for me because I am using the CLR.  For some reason the CLI doesn't update the version.

after various clumsy attempts at fixing, in the end my environment got corrupted and had to reinstall, fortunately had app backups..

M

**Comment by GarryXian** [comment]
expecting the New and improved Rest API experience in Dec

**Reply by mjashanks** [reply]
It's coming!

**Reply by joebudi** [reply]
P.s this was shipped!

**Comment by SantoshSrinivas79** [comment]
Is there any new release to Budibase @joebudi @mjashanks ?
We are on `1.0.46`

**Reply by shogunpurple** [reply]
https://github.com/Budibase/budibase/releases

**Reply by SantoshSrinivas79** [reply]
Cool! Just updated ... where can I see the running version now? `builder/portal/settings/update` doesn't show the version anymore
